File: Algorithms/StoogeSort.java
public class StoogeSort {

    // Function for stooge sort
    public static void stoogeSort(int[] arr, int l, int h) {
        // If the first element is greater than the last, swap them
        if (arr[l] > arr[h]) {
            int temp = arr[l];
            arr[l] = arr[h];
            arr[h] = temp;
        }

        // If there's more than two elements in the array
        if (h - l + 1 > 2) {
            int t = (h - l + 1) / 3;

            // Recursively sort first 2/3 of the array
            stoogeSort(arr, l, h - t);

            // Recursively sort last 2/3 of the array
            stoogeSort(arr, l + t, h);

            // Again, recursively sort first 2/3 of the array
            stoogeSort(arr, l, h - t);
        }
    }

    public static void main(String[] args) {
        // Example array
        int[] arr = {5, 7, 3, 2, 8, 1, 9};

        // Call the sorting function
        stoogeSort(arr, 0, arr.length - 1);

        // Print sorted array
        System.out.print("Sorted array: ");
        for (int i : arr) {
            System.out.print(i + " ");
        }
    }
}

File: Algorithms/Stooge_Sort.cpp
#include <iostream>
#include <algorithm>
using namespace std;

void stoogeSort(int arr[], int start, int end) {

    if (start >= end)
        return;

    if (arr[start] > arr[end])
        swap(arr[start], arr[end]);

    if (end - start + 1 > 2) {
        int third = (end - start + 1) / 3;

        stoogeSort(arr, start, end - third);

        stoogeSort(arr, start + third, end);

        stoogeSort(arr, start, end - third);
    }
}

int main() {
    int arr[] = {2, 4, 5, 3, 1};
    int n = sizeof(arr) / sizeof(arr[0]);

    cout << "Original array: ";
    for (int i = 0; i < n; i++)
        cout << arr[i] << " ";
    cout << endl;

    stoogeSort(arr, 0, n - 1);

    cout << "Sorted array: ";
    for (int i = 0; i < n; i++)
        cout << arr[i] << " ";
    cout << endl;

    return 0;
}

